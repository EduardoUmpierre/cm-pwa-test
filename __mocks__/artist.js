import { GET_ARTIST_ALBUMS } from '../src/screens/Home';

export const albums = [
  {
    name: 'Live 2012 - O2 Shepherds Bush Empire',
    id: '7K7130jJVwDhA2nQrUt5mI',
    image: 'https://i.scdn.co/image/ab67616d0000b273a042ccc97383a65cd46147e0',
  },
  {
    name: 'Live at High Voltage Festival 2011',
    id: '5C32hd4jdKdRRgpsszegwC',
    image: 'https://i.scdn.co/image/ab67616d0000b2735dffe646df71ff948b4f16b8',
  },
];

export const artist = {
  name: 'Thin Lizzy',
  id: '6biWAmrHyiMkX49LkycGqQ',
  image: 'https://i.scdn.co/image/cbebfb93be8c2ac47ef6b8a2186d11b3298abe49',
  albums,
};

export const artistMock = (queryArtists = [artist]) => ({
  request: {
    query: GET_ARTIST_ALBUMS,
    variables: { name: 'Thin Lizzy' },
  },
  result: {
    data: {
      queryArtists,
    },
  },
});

export const errorMock = () => ({
  request: {
    query: GET_ARTIST_ALBUMS,
    variables: { name: 'ABCDEF' },
  },
  error: new Error('Error'),
});

export const emptyMock = () => ({
  request: {
    query: GET_ARTIST_ALBUMS,
    variables: { name: 'ABCDEF' },
  },
  result: {
    data: {
      queryArtists: [],
    },
  },
});
