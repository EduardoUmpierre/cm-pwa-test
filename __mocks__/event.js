export default {
  preventDefault: () => null,
  target: {
    value: '',
  },
};
