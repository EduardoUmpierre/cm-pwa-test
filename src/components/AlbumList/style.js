import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-gap: 2rem;
  grid-template-columns: repeat(auto-fill, minmax(20rem, 1fr));
`;
