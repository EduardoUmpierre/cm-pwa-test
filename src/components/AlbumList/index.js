import React from 'react';
import { Container } from './style';
import AlbumListItem from '../AlbumListItem';

const AlbumList = ({ items }) => (
  <Container>
    {items.map(item => (
      <AlbumListItem key={item.id} item={item} />
    ))}
  </Container>
);

export default AlbumList;
