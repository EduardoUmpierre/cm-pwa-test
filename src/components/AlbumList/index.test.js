import React from 'react';
import { shallow, mount } from 'enzyme';
import AlbumList from '.';
import { albums } from '../../../__mocks__/artist';

describe('AlbumList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<AlbumList />);
    expect(wrapper).toBeTruthy();
  });

  it('should render the items correctly', () => {
    const wrapper = mount(<AlbumList albums={albums} />);
    expect(wrapper).toMatchSnapshot();
  });
});
