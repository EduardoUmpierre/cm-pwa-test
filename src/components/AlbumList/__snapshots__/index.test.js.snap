// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumList should render the items correctly 1`] = `ReactWrapper {}`;
